.github/workflows/docker.yml

```
@@ -0,0 +1,37 @@
+name: Docker Build and Push
+
+on:
+  workflow_dispatch:
+  release:
+    types: [published]
+
+jobs:
+  docker-build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Find tag
+        id: tagger
+        run: |
+          echo "tag=$(git fetch --prune --unshallow && git describe --tags --match 'v*')" >> $GITHUB_OUTPUT
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./service/Dockerfile
+          push: true
+          tags: spacemeshos/post-service:latest, spacemeshos/post-service:${{ GITHUB.SHA  }}, spacemeshos/post-service:${{steps.tagger.outputs.tag}}
```

You can use `github.ref_name` to get the git tag or branch that triggered the workflow, we are also using it in `go-spacemesh` to tag docker images.

```suggestion
          tags: spacemeshos/post-service:latest, spacemeshos/post-service:${{ GITHUB.SHA  }}, spacemeshos/post-service:${{ github.ref_name }}
```